test/widgets/actions_test.dart

```diff
@@ -30,19 +41,247 @@ void main() {
+  /// unregister-token request instead of reusing the store's connection.
+  /// To enable callers to prepare responses for that request, this function
+  /// creates a new [FakeApiConnection] and caches it in [TestGlobalStore]
+  /// for [unregisterToken] to pick up.
+  ///
+  /// Call this instead of just turning on
+  /// [TestGlobalStore.useCachedApiConnections] so that [unregisterToken]
+  /// doesn't try to call `close` twice on the same connection instance,
+  /// which isn't allowed. (Once by the unregister-token code
+  /// and once as part of removing the account.)
+  FakeApiConnection separateConnection() {
+    testBinding.globalStore
+      ..clearCachedApiConnections()
+      ..useCachedApiConnections = true;
+    return testBinding.globalStore
+      .apiConnectionFromAccount(eg.selfAccount) as FakeApiConnection;
+  }
+
+  String unregisterApiPathForPlatform(TargetPlatform platform) {
+    return switch (platform) {
+      TargetPlatform.android => '/api/v1/users/me/android_gcm_reg_id',
+      TargetPlatform.iOS     => '/api/v1/users/me/apns_device_token',
+      _                      => throw Error(),
+    };
+  }
+
+  void assertSingleUnregisterRequest(
+    FakeApiConnection connection, {
+    String? expectedToken,
+  }) {
+    final subject = check(connection.takeRequests()).single.isA<http.Request>()
+      ..method.equals('DELETE')
+      ..url.path.equals(unregisterApiPathForPlatform(defaultTargetPlatform));
+    if (expectedToken != null) {
+      subject.bodyFields.deepEquals({'token': expectedToken});
+    }
+  }
+
+  group('logOutAccount', () {
+    testWidgets('smoke', (tester) async {
+      await prepare(tester);
+      check(testBinding.globalStore).accountIds.single.equals(eg.selfAccount.id);
+      const unregisterDelay = Duration(seconds: 5);
+      assert(unregisterDelay > TestGlobalStore.removeAccountDuration);
+      final newConnection = separateConnection()
+        ..prepare(delay: unregisterDelay, json: {'msg': '', 'result': 'success'});
+
+      final future = logOutAccount(context, eg.selfAccount.id);
+      // Unregister-token request and account removal dispatched together
+      assertSingleUnregisterRequest(newConnection);
+      check(testBinding.globalStore.takeDoRemoveAccountCalls())
+        .single.equals(eg.selfAccount.id);
+
+      await tester.pump(TestGlobalStore.removeAccountDuration);
+      await future;
+      // Account removal not blocked on unregister-token response
+      check(testBinding.globalStore).accountIds.isEmpty();
+      check(connection.isOpen).isFalse();
+      check(newConnection.isOpen).isTrue(); // still busy with unregister-token
+
+      await tester.pump(unregisterDelay - TestGlobalStore.removeAccountDuration);
+      check(newConnection.isOpen).isFalse();
+
+      // No unexpected requests made on either connection
+      check(connection.takeRequests()).isEmpty();
+      check(newConnection.takeRequests()).isEmpty();
```

Comment:
I think these are redundant with the check in `send` — requests that are unexpected won't have a prepared response, so they'll throw, and the test framework will catch the unhandled exception and report.